src/common/database.c

```diff
@@ -1712,6 +1712,24 @@ static int _upgrade_library_schema_step(dt_database_t *db, int version)
     sqlite3_exec(db->handle, "COMMIT", NULL, NULL, NULL);
     new_version = 31;
   }
+  else if(version == 31)
+  {
+    sqlite3_exec(db->handle, "BEGIN TRANSACTION", NULL, NULL, NULL);
+
+    // remove duplicates
+    TRY_EXEC("DELETE FROM main.meta_data WHERE rowid NOT IN (SELECT min(rowid) "
```

Comment:
I see that now and indeed you took exactly the fix I proposed, that is smallest rowid. You've then double checked that this gets the displayed values, right ?

Reply:
> I see that now and indeed you took exactly the fix I proposed, that is smallest rowid. You've then double checked that this gets the displayed values, right ?

I've checked on sqlstudio with faulty data.
I've also checked with db without duplicate (and got it untouched :) ) but not doubled checked with faulty data. Give me five minutes.

Reply:
Double checked.